net/portmapper/portmapper.go

```diff
@@ -371,23 +378,35 @@ func (c *Client) createOrGetMapping(ctx context.Context) (external netaddr.IPPor
 		prevPort = m.External().Port()
 	}
 
+	if DisablePCP && DisablePMP {
+		c.mu.Unlock()
+		if external, ok := c.getUPnPPortMapping(ctx, gw, internalAddr, prevPort); ok {
+			return external, nil
+		}
+		return netaddr.IPPort{}, NoMappingError{ErrNoPortMappingServices}
+	}
+
 	// If we just did a Probe (e.g. via netchecker) but didn't
 	// find a PMP service, bail out early rather than probing
 	// again. Cuts down latency for most clients.
 	haveRecentPMP := c.sawPMPRecentlyLocked()
+	haveRecentPCP := c.sawPCPRecentlyLocked()
 
+	m := &pmpMapping{
```

Comment:
is this used for PCP as well? if not, why is only one created up here, and why PMP?

Reply:
this is from the old code, so I mainly left this in to preserve that, but PMP also requires two steps, getting the mapping and getting the external IP. I just shifted it around so that if it doesn't gets called it won't be allocated. I left a comment above it documenting the behaviour